Source/CBLRouter+Changes.m

```diff
@@ -235,21 +256,49 @@ - (void) dbChanged: (NSNotification*)n {
         if (_changesMode == kLongPollFeed) {
             [changes addObject: rev];
         } else {
-            Log(@"CBL_Router: Sending continous change chunk");
             [self sendContinuousLine: [self changeDictForRev: rev]];
         }
+        _changesSince = rev.sequence;
     }
 
-    if (_changesMode == kLongPollFeed && changes.count > 0) {
-        Log(@"CBL_Router: Sending longpoll response");
-        [self sendResponseHeaders];
-        NSDictionary* body = [self responseBodyForChanges: changes since: 0];
-        _response.body = [CBL_Body bodyWithProperties: body];
-        [self sendResponseBodyAndFinish: YES];
-    }
+    if (_changesMode == kLongPollFeed && changes.count > 0)
+        [self sendLongpollResponseForChanges: changes since: 0];
+    else
+        [self startTimeout];
 
     retainSelf = nil;
 }
 
 
+- (void)sendLongpollResponseForChanges: (NSArray*)changes since: (UInt64)since {
+    Log(@"CBL_Router: Sending longpoll response");
+    [self sendResponseHeaders];
+    NSDictionary* body = [self responseBodyForChanges: changes since: since];
+    _response.body = [CBL_Body bodyWithProperties: body];
+    [self sendResponseBodyAndFinish: YES];
+}
+
+
+- (void) startTimeout {
+    assert(_changesTimeout > 0);
+    [_changesTimeoutTimer invalidate];
+    CBL_Router* weakSelf = self;
+    _changesTimeoutTimer = [NSTimer scheduledTimerWithTimeInterval: _changesTimeout
+                                                           repeats: NO block: ^(NSTimer *timer) {
+        CBL_Router* strongSelf = weakSelf;
+        if (_changesMode == kLongPollFeed) {
+            [strongSelf sendLongpollResponseForChanges: @[] since: _changesSince];
```

Comment:
By accessing `_changesSince` the block implicitly captures `self`, making the cycle reappear … change to `strongSelf->_changesSince`.